spec/mailers/shf_mailer_spec.rb

```diff
@@ -1,5 +1,206 @@
+    end
+
+    it 'greeting uses first and last name' do
+      expect(@email).to have_body_text(I18n.t('shf_mailer.greeting', greeting_name: @recipient.full_name))
+    end
+
+
+    describe 'signoff section' do
+
+      it 'should have a signoff' do
+        expect(@email).to have_body_text(I18n.t('shf_mailer.shf_signoff'))
+      end
+
+
+      describe 'signature' do
+
+        it 'should have a signature' do
+          expect(@email).to have_body_text(I18n.t('shf_mailer.shf_signature'))
+        end
+
+        it 'should have a link to the site - text format' do
+          expect(@email.text_part.body).to match(/#{I18n.t('shf_mailer.shf_signature')}\s+#{root_url}/)
+        end
+
+        it 'should have a link to the site - html format' do
+          expect(@email.html_part.body.to_s).to match(/<div id='signature'>\s*(.)*\s*<a(.)*href="#{root_url}">#{I18n.t('shf_mailer.shf_signature_url')}<\/a>\s*<\/div>/)
+        end
+
+      end
+
+    end
+
+
+    it 'text in the footer ' do
+      expect(@email).to have_body_text(I18n.t('shf_mailer.footer.text'))
+    end
+
+  end
+
+
+  it 'language is set' do
+    #pending 'need to change locale and ensure the right language is in the email created'
+  end
+
+
+  describe 'use :mailgun delivery method' do
+
+    before(:all) { Rails.configuration.action_mailer.delivery_method = :mailgun }
+
+    before(:each) { subject.mailgun_client.enable_test_mode! }
+
+    after(:each) { subject.mailgun_client.disable_test_mode! }
+
+
+    describe 'simple test email' do
+
+      let(:test_email_response) {
+        result = subject.mail(to: 'recipient@example.com', subject: "Test email from SHF", template_name: TEST_TEMPLATE)
+        result.deliver
+      }
+
+      it 'is delivered' do
+        expect(test_email_response.message_id).not_to be_nil
+
+        delivered = ShfMailer.deliveries
+        expect(delivered.count).to eq 1
+        expect(test_email_response[:message_id]).to eq(delivered.first[:message_id])
+      end
+
+    end
+
+
+    describe 'can send with attachments' do
+
+      let(:test_email_response) {
+
+        attachment1 = {
+            content: file_fixture('diploma.pdf'),
+            mime_type: 'application/pdf',
+        }
+        attachment2 = {
+            content: file_fixture('microsoft-word.docx'),
+            mime_type: 'application/docx',
+        }
+
+        result = subject.mail(to: 'recipient@example.com',
+                              subject: 'Test email with attachment',
+                              template_name: TEST_TEMPLATE,
+                              attachments: [attachment1, attachment2])
+        result.deliver
+      }
+
+
+      it 'is delivered' do
+        expect(test_email_response.message_id).not_to be_nil
+
+        delivered = ShfMailer.deliveries
+        expect(delivered.count).to eq 1
+        expect(test_email_response[:message_id]).to eq(delivered.first[:message_id])
+      end
+
+    end
+
+
+  end
+
```

Comment:
Actually, I ended up deleting the reset-password (unit) tests PR - this PR completely tested that email as part of testing the overall email/mailgun implementation 👍